src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -0,0 +1,700 @@
+					StreamStatsWrapper streamInfoWrapper = objectMapper.convertValue(responseMap, StreamStatsWrapper.class);
+
+					// Check if converted object is not a stream
+					if (streamInfoWrapper.getStream().getStreamId() == null){
+						break;
+					}
+
+					StreamStats streamStats = streamInfoWrapper.getStreamStats();
+					Stream stream = streamInfoWrapper.getStream();
+
+					// Stream name filtering
+					if (this.streamNameFilter != null && streamsNameFiltered != null && streamsNameFiltered.contains(stream.getStreamName())) {
+						populateStreamStats(stats, streamInfoWrapper);
+						updateLocalStreamConfigInfo(streamInfoWrapper, Integer.parseInt(stream.getStreamId()));
+						continue;
+					}
+
+					// Stream status filtering
+					if (this.streamStatusFilter != null && streamsStatusFiltered != null && !streamsStatusFiltered.contains(streamStats.getState())) {
+						continue;
+					}
+
+					// Port number filtering
+					if (this.portNumberFilter != null && portNumbersFiltered != null) {
+						Integer port = Integer.parseInt(streamInfoWrapper.getStreamConfigInfo().getPort());
+						boolean isValidPort = validatePortRange(port);
+						if (!isValidPort) {
+							continue;
+						}
+					}
+					if (this.streamStatusFilter != null) {
+						populateStreamStats(stats, streamInfoWrapper);
+						updateLocalStreamConfigInfo(streamInfoWrapper, Integer.parseInt(stream.getStreamId()));
+					}
+					if (this.portNumberFilter != null) {
+						populateStreamStats(stats, streamInfoWrapper);
+						updateLocalStreamConfigInfo(streamInfoWrapper, Integer.parseInt(stream.getStreamId()));
+					}
+					if (this.streamNameFilter == null) {
+						populateStreamStats(stats, streamInfoWrapper);
+						updateLocalStreamConfigInfo(streamInfoWrapper, Integer.parseInt(stream.getStreamId().trim()));
+					}
+				}
+			} else {
+				updateStreamStatisticsFailedMonitor();
+			}
+		} catch (Exception e) {
+				logger.error("Error while retrieve stream statistics: ", e);
+			updateStreamStatisticsFailedMonitor();
+		}
+	}
+
+	/**
+	 * Update failedMonitor with getting stream statistics error message
+	 */
+	private void updateStreamStatisticsFailedMonitor() {
+		failedMonitor.put(MonitoringMetricGroup.STREAM_STATISTICS.getName(), DecoderConstant.GETTING_STREAM_STATS_ERR);
+	}
+
+	/**
+	 * This method is used update stream statistic from DTO
+	 *
+	 * @param stats list statistics property
+	 * @param stream pair of stream config and stats
+	 */
+	private void populateStreamStats(Map<String, String> stats, StreamStatsWrapper stream) {
+		String streamStatisticGroup = MonitoringMetricGroup.STREAM_STATISTICS.getName() + stream.getStream().getStreamName() + DecoderConstant.HASH;
+
+		for (StreamMonitoringMetric item : StreamMonitoringMetric.values()) {
+			if (stream.getStream() != null) {
+				stats.put(streamStatisticGroup + item.getName(), getDefaultValueForNullData(stream.getStream().getValueByStreamMonitoringMetric(item)));
```

Comment:
streamStatisticGroup + item.getName() can be extracted into a variable and reused